consensus/src/chained_bft/chained_bft_smr.rs

```diff
@@ -256,20 +249,60 @@ impl<T: Payload> StateMachineReplication for ChainedBftSMR<T> {
             safety_rules,
             state_computer,
             txn_manager,
-            self.network.clone(),
+            network.clone(),
             Arc::clone(&self.storage),
             time_service.clone(),
             true,
-            Arc::clone(&self.epoch_mgr),
+            epoch_mgr.validators(),
         );
 
-        self.start_event_processing(
-            executor,
-            event_processor,
-            timeout_receiver,
-            network_receivers,
-        );
+        Self::start_event_processing(executor, event_processor, timeout_receiver, network);
+    }
+}
+
+impl<T: Payload> StateMachineReplication for ChainedBftSMR<T> {
+    type Payload = T;
+
+    /// We're following the steps to start
+    /// 1. Align initial data with libradb (sync if necessary) (We need initial trusting peers to connect to)
+    /// 2. Construct the EpochManager from the latest libradb state
```

Comment:
I agree that it would be nicer to move 1&2 up out of here as the main thing is to provide a way to start a new epoch and we want to have a ChainedBftSMR instance correspond to one epoch.